### DISCUSSION TITLE
Failed to publish AppStore "Asset validation failed (90512) Invalid sdk value."

### DISCUSSION THREAD

**Discussion by unbam** [codemagic-ci-cd/codemagic-docs#2839] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

## env
```
== Use Xcode 16.2 ==
> sudo xcode-select --switch /Applications/Xcode-16.2.app/Contents/Developer
Xcode 16.2
Build version 16C5013f

== Use CocoaPods version 1.16.2 ==
> gem uninstall cocoapods --executables --all
Removing pod
Removing sandbox-pod
Successfully uninstalled cocoapods-1.15.2

> gem install cocoapods --version 1.16.2
Successfully installed cocoapods-core-1.16.2
Successfully installed cocoapods-1.16.2
Parsing documentation for cocoapods-core-1.16.2
Installing ri documentation for cocoapods-core-1.16.2
Parsing documentation for cocoapods-1.16.2
Installing ri documentation for cocoapods-1.16.2
Done installing documentation for cocoapods-core, cocoapods after 1 seconds
2 gems installed

> gem list '^cocoapods$' --no-verbose
cocoapods (1.16.2)

== Use Java version 17 ==
> java -version
openjdk version "17.0.13" 2024-10-15 LTS
OpenJDK Runtime Environment Zulu17.54+21-CA (build 17.0.13+11-LTS)
OpenJDK 64-Bit Server VM Zulu17.54+21-CA (build 17.0.13+11-LTS, mixed mode, sharing)

== Codemagic CLI tools ==
> codemagic-cli-tools version
codemagic-cli-tools 0.54.4
```

## flutter

```
Flutter 3.24.5 • Dart 3.5.4
```

## Distribution
iOS code signing[enabled]
App Store Connect[enabled]

### Expected results

UPLOAD SUCCEEDED

### Actual results

> Upload "/Users/builder/clone/app/build/ios/ipa/xxxxx.ipa" to App Store Connect
Running altool at path '/Applications/Xcode-16.2.app/Contents/SharedFrameworks/ContentDeliveryServices.framework/Frameworks/AppStoreService.framework/Support/altool'...
Running altool at path '/Applications/Xcode-16.2.app/Contents/SharedFrameworks/ContentDeliveryServices.framework/Frameworks/AppStoreService.framework/Support/altool'...
2024-11-19 01:09:18.908 *** Error: [ContentDelivery.Uploader.xxxxxxxxxxxx] Asset validation failed (90512) Invalid sdk value. The value provided for the sdk portion of LC_BUILD_VERSION in Runner.app/Frameworks/App.framework/App is 18.2 which is greater than the maximum allowed value of 18.1. (ID: xxxxxxxx-xxxx-xxxx-xxxxxxxxxxxx)
2024-11-19 01:09:19.064 [ContentDelivery.Uploader.xxxxxxxxxxxx] 
=============
UPLOAD FAILED with 1 error.
=============
{"tool-version":"8.003.16003","tool-path":"\/Applications\/Xcode-16.2.app\/Contents\/SharedFrameworks\/ContentDeliveryServices.framework\/Versions\/A\/Frameworks\/AppStoreService.framework","os-version":"15.1.0","product-errors":[{"message":"Asset validation failed","userInfo":{"NSUnderlyingError":"Error Domain=IrisAPI Code=-19241 \"Asset validation failed\" UserInfo={status=409, detail=Invalid sdk value. The value provided for the sdk portion of LC_BUILD_VERSION in Runner.app\/Frameworks\/App.framework\/App is 18.2 which is greater than the maximum allowed value of 18.1., id=xxxxxxxx-xxxx-xxxx-xxxxxxxxxxxx, <br/>code=STATE_ERROR.VALIDATION_ERROR.90512, title=Asset validation failed, NSLocalizedFailureReason=Invalid sdk value. The value provided for the sdk portion of LC_BUILD_VERSION in Runner.app\/Frameworks\/App.framework\/App is 18.2 which is greater than the maximum allowed value of 18.1., NSLocalizedDescription=Asset validation failed}","NSLocalizedDescription":"Asset validation failed","iris-code":"STATE_ERROR.VALIDATION_ERROR.90512","NSLocalizedFailureReason":"Invalid sdk value. The value provided for the sdk portion of LC_BUILD_VERSION in Runner.app\/Frameworks\/App.framework\/App is 18.2 which is greater than the maximum allowed value of 18.1. (ID: xxxxxxxx-xxxx-xxxx-xxxxxxxxxxxx)"},"code":90512}]}

**Comment by unbam** [comment]
Resolved :)
I changed from Xcode 16.2 to Xcode 16.1

**Reply by jusevasa** [reply]
@unbam Thank you so much 💯 !!!

**Comment by sglim** [comment]
More detail:

<img width="699" alt="image" src="https://github.com/user-attachments/assets/5c5f6788-d280-4971-99ec-77f4507d8b42">

- Go to Workflow settings
- Go to the Environment section
- Change the Xcode version from 16.2 Beta -> 16.1

I think Apple release Xcode beta on the Xcodecloud by mistake.

![image](https://github.com/user-attachments/assets/db891300-237a-499c-bc19-6df8337c82d8)

**Reply by large** [reply]
Someone at Apple has messed up the "latest release" vs "latest beta or release".
Thank you for sharing!

**Reply by ihijazi** [reply]
The whole App Connect was wild yesterday! Getting same error as this, thanks for the thumbs up.

**Reply by ihijazi** [reply]
Still XCode Cloud is shaky today on XCode IDE as well, keeps logging me out.
